service/matching/matching_engine.go

```diff
@@ -1653,10 +1654,14 @@ pollLoop:
 			TaskId:      task.nexus.taskID,
 		}
 		serializedToken, _ := e.tokenSerializer.SerializeNexusTaskToken(taskToken)
+
+		nexusReq := task.nexus.request.GetRequest()
+		nexusReq.Header[nexus.HeaderRequestTimeout] = time.Until(task.nexus.deadline).String()
```

Comment:
I'd put a dedicated field on the proto for this instead of using `Header`.
We're also not propagating `CallbackHeader` as expected BTW, we can address both issues at once.

Reply:
Hmm.. you know what, I think it's fine the way it is.
Don't have a super strong opinion here.